k256/src/arithmetic/mul.rs

```diff
@@ -371,6 +372,41 @@ fn lincomb_generic<const N: usize>(xs: &[ProjectivePoint; N], ks: &[Scalar; N])
     acc
 }
 
+static GEN_LOOKUP_TABLE: Lazy<[LookupTable; 33]> = Lazy::new(precompute_gen_lookup_table);
+
+fn precompute_gen_lookup_table() -> [LookupTable; 33] {
+    let mut gen = ProjectivePoint::GENERATOR;
+    let mut res = [LookupTable::default(); 33];
+    #[allow(clippy::needless_range_loop)]
+    for i in 0..33 {
+        res[i] = LookupTable::from(&gen);
+        // We are storing tables spaced by two radix steps,
+        // to decrease the size of the precomputed data.
+        for _ in 0..8 {
+            gen = gen.double();
+        }
+    }
+    res
+}
+
+/// Calculages `k * G`, where `G` is the generator.
+pub fn mul_by_generator(k: &Scalar) -> ProjectivePoint {
+    let digits = Radix16Decomposition::<65>::new(k);
+    let table = *GEN_LOOKUP_TABLE;
+    let mut acc = table[32].select(digits.0[64]);
+    let mut acc2 = ProjectivePoint::IDENTITY;
+    for i in (0..32).rev() {
+        acc2 += &table[i].select(digits.0[i * 2 + 1]);
+        acc += &table[i].select(digits.0[i * 2]);
+    }
+    // This is the price of halving the precomputed table size (from 60kb to 30kb)
+    // The performance hit is minor, about 3%.
+    for _ in 0..4 {
+        acc2 = acc2.double();
+    }
+    acc + acc2
+}
```

Comment:
@fjarri one thing I noticed too late refactoring... I think this would make sense as an inherent method of `ProjectivePoint` so it can be called as `ProjectivePoint::mul_by_generator(&scalar)`?

That way you don't have to import it separately.

Reply:
Yes, I think I put it into `mul.rs` to make use of `Radix16Decomposition` and `LookupTable`, but it can be definitely defined in an `impl` block there.

Reply:
Cool, I think I'll go ahead and refactor it into an inherent method in #705 then